pkg/cmd/step/create/step_create_meta_pipeline.go

```diff
@@ -0,0 +1,325 @@
+	cmd.Flags().StringVar(&options.PullRefs, pullRefOptionName, "", "The Prow pull ref specifying the references to merge into the source")
+	cmd.Flags().StringVarP(&options.PipelineKind, pipelineKindOptionName, "k", "", "The kind of pipeline to create such as: "+strings.Join(jenkinsfile.PipelineKinds, ", "))
+
+	cmd.Flags().StringArrayVarP(&options.CustomLabels, labelOptionName, "l", nil, "List of custom labels to be applied to the generated PipelineRun (can be use multiple times)")
+	cmd.Flags().StringArrayVarP(&options.CustomEnvs, "env", "e", nil, "List of custom environment variables to be applied to resources that are created (can be use multiple times)")
+	// TODO issue #3342 what is the long term strategy here? Should this be handled via env variables? Are CLI flags a good idea? What's about build reproducibility
+	cmd.Flags().StringVar(&options.DefaultImage, "default-image", syntax.DefaultContainerImage, "Specify the docker image to use if there is no image specified for a step")
+
+	cmd.Flags().StringVarP(&options.Context, "context", "c", "", "The pipeline context if there are multiple separate pipelines for a given branch")
+	cmd.Flags().StringVar(&options.Revision, "revision", "", "The git revision to checkout, can be a branch name or git sha")
+	cmd.Flags().StringVar(&options.PullRequestNumber, "pr-number", "", "If a Pull Request this is it's number")
+	cmd.Flags().StringVar(&options.SourceName, "source", "source", "The name under which to checkout the repository. Defaults to 'source'")
+	cmd.Flags().StringVarP(&options.Trigger, "trigger", "t", string(pipelineapi.PipelineTriggerTypeManual), "The kind of pipeline trigger")
+	cmd.Flags().StringVar(&options.ServiceAccount, "service-account", "tekton-bot", "The Kubernetes ServiceAccount to use to run the pipeline")
+
+	cmd.Flags().StringVarP(&options.OutDir, "output", "o", "out", "The directory to write the output to as YAML")
+	cmd.Flags().BoolVar(&options.NoApply, "no-apply", false, "Disables creating the pipeline resources in the kubernetes cluster and just outputs the generated Task to the console or output file")
+	cmd.Flags().BoolVar(&options.ViewSteps, "view", false, "Just view the steps that would be created")
+
+	options.AddCommonFlags(cmd)
+	return cmd
+}
+
+// Run implements this command
+func (o *StepCreatePipelineOptions) Run() error {
+	err := o.validateCommandLineFlags()
+	if err != nil {
+		return err
+	}
+
+	gitInfo, _ := gits.ParseGitURL(o.SourceURL)
+
+	tektonClient, jxClient, kubeClient, ns, err := o.getClientsAndNamespace()
+	if err != nil {
+		return err
+	}
+
+	podTemplates, err := o.getPodTemplates(kubeClient, ns, apps.AppPodTemplateName)
+	if err != nil {
+		return errors.Wrap(err, "unable to retrieve pod templates")
+	}
+
+	pipelineResourceName := tekton.PipelineResourceName(gitInfo, o.Branch, o.Context)
+	buildNumber, err := tekton.GenerateNextBuildNumber(tektonClient, jxClient, ns, gitInfo, o.Branch, retryDuration, pipelineResourceName)
```

Comment:
Is this going to result in build number 1 for the metapipeline and build number 2 for the generated pipeline?

Reply:
...ideally they'd be the same number. Which I _think_ they will be. That'll let us use the `PipelineActivity.Pipeline` and `PipelineActivity.Build` to find both the metapipeline `PipelineRun` and the generated pipeline `PipelineRun` (based on labels on the `Pipeline`/`PipelineRun`/`Pod`/etc) for the purposes of updating the `PipelineActivity` and running `jx get build logs`. I think.